src/mmw/js/src/analyze/models.js

```diff
@@ -54,27 +54,37 @@ var AnalyzeTaskModel = coreModels.TaskModel.extend({
     }
 });
 
-AnalyzeTaskModel.getSingleton = function(App, aoi) {
-    if (!App.analyzeModel) {
-        App.analyzeModel = createTaskModel(JSON.stringify(aoi));
+var AnalyzeTaskCollection = Backbone.Collection.extend({
+    model: AnalyzeTaskModel
+});
+
+AnalyzeTaskCollection.getSingleton = function(App, aoi) {
+    if (!App.analyzeCollection) {
+        App.analyzeCollection = new AnalyzeTaskCollection([
+            { area_of_interest: aoi, taskName: 'analyze' },
+            { area_of_interest: aoi, taskName: 'analyze/animals' },
+            { area_of_interest: aoi, taskName: 'analyze/pointsource' },
+        ]);
     }
 
-    return App.analyzeModel;
+    return App.analyzeCollection;
 };
 
-// Pass in the serialized Area of Interest for
-// caching purposes (_.memoize returns the same
-// results for any object), and deserialize
-// the AoI for use on the model.
-var createTaskModel = _.memoize(function(aoi) {
-    return new AnalyzeTaskModel({
-        area_of_interest: JSON.parse(aoi)
-    });
-});
+function createAnalyzeResultViewModelCollection(App, aoi) {
```

Comment:
The only reason the first parameter is `App` is because we want to get a singleton copy of the task runners, which query and store the data from the API. These are attached to the App, which is our global scope. The logic of this function does not require App at all, and we could just as easily hand it a list of task runners. The reason I haven't done that is then we'd have to get that list outside of this function _every time_, and so we might as well make that transparent by doing it inside.